src/main/java/de/tum/in/www1/artemis/service/ResultService.java

```diff
@@ -55,6 +55,13 @@ public Result findOneWithSubmission(long id) {
             .orElseThrow(() -> new EntityNotFoundException("Result with id: \"" + id + "\" does not exist"));
     }
 
+    public void setAssessor(Result result){
+        User currentUser = userService.getUserWithGroupsAndAuthorities();
```

Comment:
I think you can get the user easier, e.g. with the principal. This might include a database call

Reply:
I did take a look into this and the Principle i get injected in the Controller does not include our User database object. An alternative using the Principle would avoid specifically looking up the current logged in user in the SpringSecurityContext but still require a database query to find that User if im not mistaken.